src/adagrams.js

```diff
@@ -1,15 +1,173 @@
+const SIZE_OF_DRAW = 10;
+const LETTER_VALUES = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+};
+
+const genPoolLetters = (poolRulesObj) => {
+  const myPool = [];
+  for (const letter in poolRulesObj) {
+    for (let i = 0; i < poolRulesObj[letter]; ++i) {
+      myPool.push(letter);
+    }
+  }
+  return myPool;
+};
+
+//Gets a random number excluding the maxNum
+const getRandomNumber = (maxNum) => {
+  return Math.floor(Math.random() * maxNum);
+};
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const poolRules = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1,
+  };
+  const draw = [];
+  const poolLetters = genPoolLetters(poolRules);
+
+  while (draw.length < SIZE_OF_DRAW) {
+    let randNum = getRandomNumber(poolLetters.length);
+    draw.push(poolLetters[randNum]);
+    poolLetters.splice(randNum, 1);
+  }
+  return draw;
+};
+
+const createLettersInHandDict = (lettersInHandArray) => {
+  const lettersDict = {};
+  for (let i = 0; i < lettersInHandArray.length; ++i) {
+    let char = lettersInHandArray[i];
+    if (lettersDict[char] === undefined) {
+      lettersDict[char] = 1;
+    } else {
+      lettersDict[char]++;
+    }
+  }
+  return lettersDict;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  if (!input || lettersInHand === []) {
+    return false;
+  }
+  const letterBank = createLettersInHandDict(lettersInHand);
+  input = input.toUpperCase();
+  for (let i = 0; i < input.length; ++i) {
+    let letter = input[i];
+    if (letterBank[letter] == undefined) {
+      return false;
+    }
+    if (letterBank[letter] == 0) {
+      return false;
+    }
+    letterBank[letter]--;
+  }
+  return true;
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  if (!word) {
+    return 0;
+  }
+  let points = 0;
+  word = word.toUpperCase();
+  for (let i = 0; i < word.length; ++i) {
```

Comment:
If we won't be reassigning the loop variable in the body of a loop, I recommend using `const` over `let` to declare the variable.